collect_app/src/main/java/org/odk/collect/android/tasks/DeleteFormsTask.java

```diff
@@ -96,15 +95,16 @@ public void setDeleteListener(DeleteFormsListener listener) {
         dl = listener;
     }
 
-    public void setContentResolver(ContentResolver resolver) {
-        cr = resolver;
-    }
-
     public int getDeleteCount() {
         return successCount;
     }
 
     public int getToDeleteCount() {
         return toDeleteCount;
     }
+
+    public void setRepositories(FormsRepository formsRepository, InstancesRepository instancesRepository) {
```

Comment:
Why not use parameter args to guarantee these are set? This seems dangerous.

Reply:
I was following the original structure which used setters to get the dependencies in. You're right though that we shoud improve in that while we're here!